### PR TITLE
fix: [PIE-2626]: Set failure strategies to all errors in approvals an…

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
Summary:

Set failure strategies to "AllErrors"
Sync Approval stage UI with yaml
Updated test case snapshot

 Jira Links:

https://harness.atlassian.net/browse/PIE-2626

Screenshots:
Screenshot 2022-03-07 at 1 43 44 PM
Screenshot 2022-03-07 at 1 43 53 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by goyalsonali** [comment]
retrigger titlecheck

**Comment by goyalsonali** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineStudio/FailureStrategy/FailureStrategy.tsx [hunk 1]
```diff
@@ -83,24 +83,32 @@ export function FailureStrategy(props: FailureStrategyProps, ref: StepCommandsRe
     }
   }))
 
+  useEffect(() => {
+    debouncedUpdate({ failureStrategies: getInitialValues() })
```
Comment (vkbansal) [inline src/modules/70-pipeline/components/PipelineStudio/FailureStrategy/FailureStrategy.tsx:87]: Why is this required?

Comment (goyalsonali) [reply src/modules/70-pipeline/components/PipelineStudio/FailureStrategy/FailureStrategy.tsx:87]: There is done so as to sync FailureStrategies with yaml for ApprovalStage.

File: src/modules/70-pipeline/components/PipelineStudio/FailureStrategy/FailureStrategy.tsx [hunk 2]
```diff
@@ -83,24 +83,32 @@ export function FailureStrategy(props: FailureStrategyProps, ref: StepCommandsRe
...
+  }, [])
```
Comment (diptiman-adak) [inline src/modules/70-pipeline/components/PipelineStudio/FailureStrategy/FailureStrategy.tsx:88]: Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'debouncedUpdate' and 'getInitialValues'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232232)
```suggestion
  }, [debouncedUpdate, getInitialValues])
```
